pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -117,9 +117,16 @@ func configureTracingFromTelemetry(
 		// something like: configureFromProxyConfig(tracingCfg, opts.proxy.Metadata.ProxyConfig.Tracing)
 	}
 
-	// gracefully fallback to MeshConfig configuration. It will act as an implicit
-	// parent configuration during transition period.
-	configureSampling(h.Tracing, spec.RandomSamplingPercentage)
+	var sampling float64
+	if spec.RandomSamplingPercentage != nil {
+		sampling = *spec.RandomSamplingPercentage
```

Comment:
I doubt when m.RandomSamplingPercentage is unset, with above change
 `spec.RandomSamplingPercentage = ptr.Of(m.RandomSamplingPercentage.GetValue())`

spec.RandomSamplingPercentage is always not nil, so the sampling is 0, so it can never fallback to below else branch

Reply:
when there's no Telemetry API. CRs, only default providers.

Also, should fallback to defaultConfig, when you didn't specify sample rating in Telemetry API CR.

Reply:
ok, IC.